### PR TITLE
BDOG 168 Service Page / Code and Build / CI Links Incorrect

### PR DESCRIPTION
**Pull request by chotai**
This service retrieves the team name from Github. Builds are structured under `job/$jobTeamName/job/$repo` in Jenkins. The build urls will only work when the github team name and the team name as it appears in Jenkins are the same. The `VAT View and Change` team appear as the `VATVC` team in Github so links won't work for this team for example.

Ci Open and Ci Close have been replaced with `http://build.tax.service.gov.uk`

- We no longer get the `ci url` from the config
- When a service or library belong to one team we want the link to appear as `Build`
- When a service or library belong to more then one team we want the link to appear as `Team Name Build`
- Where the url contains a space this should be url encoded correctly

### CODE REVIEW

File: test/uk/gov/hmrc/teamsandrepositories/GitRepositorySpec.scala [hunk 1]
```diff
@@ -50,43 +38,51 @@ class GitRepositorySpec extends WordSpec with Matchers with OptionValues {
...
+        Link("a-team Build", "a-team Build", "https://build.tax.service.gov.uk/job/a-team/job/a-library"),
+        Link(
+          "another-team Build",
+          "another-team Build",
+          "https://build.tax.service.gov.uk/job/another-team/job/a-library")
+      )
     }
 
-    "create links for private libraries" in {
+    "not create ci links for libraries belonging to no teams" in {
```
Comment (IanMcShane) [inline test/uk/gov/hmrc/teamsandrepositories/GitRepositorySpec.scala]: The sentence describing the scenario is a little confusing due to the "not" at the beginning.  Could you reword it to something that reads a little easier such as: "libraries without a team should not generate any links"

Comment (chotai) [reply test/uk/gov/hmrc/teamsandrepositories/GitRepositorySpec.scala]: It's intended to be read as
```
"repoGroupToRepositoryDetails" should {
"not create ci links for libraries belonging to no teams" in {
```

File: app/uk/gov/hmrc/teamsandrepositories/controller/model/RepositoryDetails.scala [hunk 2]
```diff
@@ -59,14 +58,22 @@ object RepositoryDetails {
...
+  }
 
-  private def buildUrls(repo: GitRepository, templates: Seq[UrlTemplate]) =
-    templates.map(t => Link(t.name, t.displayName, t.url(repo.name))).toList
+  private def buildCiUrl(
+    linkName: String,
+    linkDisplayName: String,
+    jobTeamName: String,
+    repoName: String): Option[Link] =
+    Try(
```
[the remark this answers is not in the document]
Comment (chotai) [reply app/uk/gov/hmrc/teamsandrepositories/controller/model/RepositoryDetails.scala]: This _should_ hopefully never throw an exception. `java.net.URI(...)` can return a `URISyntaxException` so we put this in to catch something very unexpected. Such as a malformed team name being returned from Mongo.

Perhaps it makes sense to keep it?
